README.md

```diff
@@ -53,66 +79,93 @@ Point tags are key-value pairs (strings) that are associated with a point. Point
+| aws.lambda.wf.mem.used            | Metric        | The memory used by the Lambda function in megabytes.                    |
+| aws.lambda.wf.mem.percentage      | Metric        | The percentage of memory used by the Lambda function.                   |
 
-You can send custom business metrics to Wavefront using the [go-metrics-wavefront](https://github.com/wavefrontHQ/go-metrics-wavefront) client. The below code reports a _counter_, a _delta counter_, and two _gauges_. All metric names should be unique. If you have metrics that you want to track as both _counter_ and _delta counter_, you'll have to add a suffix to one of the metrics. Having the same metric name for any two types of metrics will result in only one time series at the server and thus cause collisions.
+### Custom Metrics
 
-The code below imported this module and wrapped the _handler_ function argument in _main_ with `wflambda.Wrapper(handler)`. During each execution four metrics are collected and sent to Wavefront with both the [standard point tags](#standard-point-tags) and the point tags created in the handler.
+You can send custom business metrics to Wavefront using the `RegisterMetric()` or `RegisterCounter()` methods. Counters are values that are aggregated at the Wavefront server (like the number of invocations and metrics are pretty much every other numerical value you want to send in.
 
 ```go
 package main
 
 import (
+	"github.com/aws/aws-lambda-go/events"
 	"github.com/aws/aws-lambda-go/lambda"
-	"github.com/rcrowley/go-metrics"
-	wavefront "github.com/wavefronthq/go-metrics-wavefront"
-	wflambda "github.com/wavefronthq/wavefront-lambda-go"
+	wflambda "github.com/retgits/wavefront-lambda-go" // Import this library
 )
 
-func handler() {
-	// Point Tags
-	appTags := map[string]string{
-		"key2":   "val1",
-		"key1":   "val2",
-		"key0":   "val0",
-		"key4":   "val4",
-		"key3":   "val3",
-	}
-
-	// Register Counter with desired tags.
-	customRawCounter := metrics.NewCounter()
-	wavefront.RegisterMetric("counter", customRawCounter, appTags)
-	customRawCounter.Inc(1)
-
-	// Register Delta Counter with desired tags.
-	customDeltaCounter := metrics.NewCounter()
-	deltaCounterName := wavefront.DeltaCounterName("deltaCounter")
-	wavefront.RegisterMetric(deltaCounterName, customDeltaCounter, appTags)
-	customDeltaCounter.Inc(1)
-
-	// Register Gauge with desired tags.
-	gaugeValue := metrics.NewGauge()
-	wavefront.RegisterMetric("gaugeValue", gaugeValue, appTags)
-	gaugeValue.Update(551)
-
-	// Register Float Gauge with desired tags.
-	gaugeFloatValue := metrics.NewGaugeFloat64()
-	wavefront.RegisterMetric("gaugeFloatValue", gaugeFloatValue, appTags)
-	gaugeFloatValue.Update(551.4)
+var wfAgent = wflambda.NewWavefrontAgent(&wflambda.WavefrontConfig{})
+
+func handler() (string, error){
+	// Register a new Delta Counter
+	wfAgent.RegisterCounter("MyAwesomeCounter", float64(1))
```

Comment:
We should also add support to add metric level point tags, meaning point tags applicable to only this metric but not all metrics reported by the agent. (This was supported in the earlier library)

Reply:
If I read the source code of the wavefront lib correctly (https://github.com/wavefrontHQ/go-metrics-wavefront/blob/master/reporting/reporter.go#L342), the metric specific tags are not combined with the standard point tags. In other words, if you supply metric specific tags you're not getting the standard ones. Is that correct?

Reply:
It reports all tags. 
The metric specific tags are actually encoded with the metric name before registering with the underlying "rcrowley/go-metrics.registry". See https://github.com/wavefrontHQ/go-metrics-wavefront/blob/master/reporting/reporter.go#L343-L344

These are decoded back into tags before reporting the metric to wavefront at https://github.com/wavefrontHQ/go-metrics-wavefront/blob/3fca929deab620bfd77b9749a5f0cf221c10e62b/reporting/reporter.go#L211